include/inria_wbc/robot_dart/damages.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef IWBC_ROBORT_DART_DAMAGES_HPP_
+#define IWBC_ROBORT_DART_DAMAGES_HPP_
+
+#include <inria_wbc/exceptions.hpp>
+#include <robot_dart/robot.hpp>
+#include <robot_dart/robot_dart_simu.hpp>
+#include <robot_dart/sensor/sensor.hpp>
+
+namespace inria_wbc {
+    namespace robot_dart {
+
+        static constexpr int CUT = 0;
+        static constexpr int LOCKED = 1;
+        static constexpr int PASSIVE = 2;
+        static constexpr int MASS = 3;
+        static constexpr int LIMIT = 4;
+
+        class RobotDamages {
+        public:
+            struct damage {
+                int damage_type;
+                std::vector<std::string> affected_joints;
+                std::vector<std::shared_ptr<::robot_dart::sensor::Sensor>> affected_sensors;
+            };
+
+            RobotDamages(const std::shared_ptr<::robot_dart::Robot>& robot,
+                const std::shared_ptr<::robot_dart::RobotDARTSimu>& simu,
+                const std::vector<std::string> controllable_dofs,
+                const std::vector<std::string> all_dofs)
+                : _robot(robot), _simu(simu), _controllable_dofs(controllable_dofs), _all_dofs(all_dofs)
+            {
+                _damaged = false;
+                _sensors = _simu->sensors();
+                _active_joints_mask.resize(_all_dofs.size());
+                _active_joints_mask.setConstant(1.0); //1 if joint in robot, 0 if damaged
+                _active_dofs = _all_dofs;
+                _active_dofs_controllable = _controllable_dofs;
+            }
+
+            void cut(const std::string& link_name)
+            {
+                auto vec = _robot->body_names();
+                if (std::find(vec.begin(), vec.end(), link_name) == vec.end())
+                    IWBC_ERROR("cut ", link_name, " is not a valid robot body name");
+
+                _damaged = true;
+                std::vector<std::string> affected_joints;
+                std::vector<std::shared_ptr<::robot_dart::sensor::Sensor>> affected_sensors;
+
+                auto body_node = _robot->skeleton()->getBodyNode(link_name);
+                auto tmp_skel = body_node->split("tmp");
+
+                for (size_t i = 0; i < tmp_skel->getNumJoints(); ++i) {
+                    auto name = tmp_skel->getJoint(i)->getName();
+                    _removed_joints.push_back(name);
+                    affected_joints.push_back(name);
+                    update_sensor(name, affected_sensors);
+                }
+
+                _robot->update_joint_dof_maps();
+                update_joint_lists();
+
+                damage cut_damage = {CUT, affected_joints, affected_sensors};
+                _damage_list.push_back(cut_damage);
+            }
+
+            void motor_damage(const std::string& joint_name, const int& damage_type)
```

Comment:
we should not pass const int& but int in these cases (the reference is bigger / same size as int)

Reply:
thanks this is fixed now